app/assets/javascripts/table-select-all.es6

```diff
@@ -13,6 +13,14 @@ $(document).ready(function() {
       }
     }
 
+    function setCategory(row) {
+      if (row.find('input:checked').length) {
+        row.find(".table-category-vacant").addClass("table-category-preview");
+      } else {
+        row.find(".table-category-vacant").removeClass("table-category-preview");
+      }
```

Comment:
Try not to repeat the same code with each condition of a conditional. It means that there are two things that need to be changed and kept in step - when there could just be one:
```javascript
let vacantItem = row.find(".table-category-vacant")
if (row.find('input:checked').length) {
  vacantItem.addClass("table-category-preview");
} else {
  vacantItem.removeClass("table-category-preview");
}
```
Now if the class that identifies the vacant item needs to change - it only needs to be changed in one place, and it is much more obvious that the different behaviours are being applied to the same object.